packages/dynamite/dynamite_runtime/lib/src/dynamite_client.dart

```diff
@@ -466,7 +442,7 @@ class DynamiteClient {
     if (validStatuses?.contains(response.statusCode) ?? true) {
       return response;
     } else {
-      throw await DynamiteApiException.fromResponse(response);
+      throw DynamiteStatusCodeException(response.statusCode);
```

Comment:
It's usually very helpful to also see what the body (and headers) are for debugging purposes.

Reply:
But for the body we'd need to await the entire response and there is no reason to have the code wait.

Reply:
Please add it regardless, otherwise debugging will be a pain.

Reply:
Can we make it only do that in debug mode?

Reply:
Sure

Reply:
How do you like this approach?
I'm not that sure myself.